### PR TITLE
mon/OSDMonitor: Remove creating_pgs direct access

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -831,8 +831,8 @@ void OSDMonitor::prime_pg_temp(
   const OSDMap& next,
   pg_t pgid)
 {
-  // TODO: remove this creating_pgs direct access?
-  if (creating_pgs.pgs.count(pgid)) {
+  std::map<pg_t, std::pair<epoch_t, utime_t>> pgs;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: this is buggy. creating_pgs is not equivalent to an empty `pgs` here.

**Comment by tchaikov** [comment]
and again, please avoid using `[cleanup]` as a part the prefix.

**Comment by shinobu-x** [comment]
test this please

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -831,8 +831,7 @@ void OSDMonitor::prime_pg_temp(
   const OSDMap& next,
   pg_t pgid)
 {
-  // TODO: remove this creating_pgs direct access?
-  if (creating_pgs.pgs.count(pgid)) {
+  if (get_creating_pgs_size(pgid)) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: there are more places where we are referencing `creating_pgs.pg` in this class. i don't think removing this direct access helps in the very case helps. neither do i think we need to do so. as `creating_pgs_t` is pretty much a POD data class.

**Comment by shinobu-x** [comment]
@tchaikov Thanks for your explanation. I'm clear enough. Can we remove `// TODO: remove this creating_pgs direct access?`?

**Pull request closed without merging**